src/Diagnostics.DataProviders/ConfigurationFactory.cs

```diff
@@ -24,6 +27,17 @@ public AppSettingsDataProviderConfigurationFactory(IHostingEnvironment env)
                 .SetBasePath(Directory.GetCurrentDirectory())
                 .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
 
+            var tokenProvider = new AzureServiceTokenProvider();
+            var keyVaultClient = new KeyVaultClient(
+                new KeyVaultClient.AuthenticationCallback(
+                    tokenProvider.KeyVaultTokenCallback
+                )
+            );
+
+            builder.AddAzureKeyVault("https://appservicediagnostics.vault.azure.net/",
```

Comment:
What subscription is this key vault in?

Reply:
it is currently hosted under my test subscription but as soon as I get an AME account I plan to move it to the new subscription.

Reply:
Talking to Shekhar the dev secrets will need to be in someones dev private subscription so I'm just going to leave the developer subscription in my subscription. everyone in the team will have access via a security group though.